### PR TITLE
(CDAP-6904) Moved Privileges Proxy Cache and related service to Master.

### PR DESCRIPTION
**Pull request by bdmogal**
Jira: [CDAP-6904](https://issues.cask.co/browse/CDAP-6904)
Build: http://builds.cask.co/browse/CDAP-DUT4625

### CODE REVIEW

**Comment by anwar6953** [comment]
Invalidation of cache will happen as a part of the PR for https://issues.cask.co/browse/CDAP-6825?

**Comment by bdmogal** [comment]
yeah, this is move only. want to do smaller PRs.

**Comment by bdmogal** [comment]
This should also fix https://issues.cask.co/browse/CDAP-6943

File: cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizationEnforcementModule.java [hunk 1]
```diff
@@ -48,7 +48,16 @@ protected void configure() {
           .in(Scopes.SINGLETON);
         // bind AuthorizationEnforcer to AuthorizationEnforcementService
         bind(AuthorizationEnforcer.class).to(AuthorizationEnforcementService.class).in(Scopes.SINGLETON);
+
+        bind(PrivilegesFetcherProxyService.class).to(DefaultPrivilegesFetcherProxyService.class)
+          .in(Scopes.SINGLETON);
         bind(PrivilegesFetcher.class).to(AuthorizerAsPrivilegesFetcher.class).in(Scopes.SINGLETON);
+        bind(PrivilegesFetcher.class)
+          .annotatedWith(Names.named(PRIVILEGES_FETCHER_PROXY_CACHE))
+          .to(PrivilegesFetcherProxyService.class);
```
Comment (anwar6953) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizationEnforcementModule.java:57]: rename the class to *Cache.java?

Comment (anwar6953) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizationEnforcementModule.java:57]: the caching is done internal to which class? `PrivilegesFetcherProxyService`?

Comment (bdmogal) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizationEnforcementModule.java:57]: Yes. There are two caches. An enforcement cache is DefaultAuthorizationEnforcementCache. This is the one that's started used in master as well as system and program containers for enforcement. Internally, it uses a PrivilegesFetcher to update its cache. When the enforcement cache runs in the master, the PrivilegesFetcher used is the configured Authorizer, that talks directly to Sentry. When enforcement cache runs inside some other container, the PrivilegesFetcher is bound to a RemotePrivilegesFetcher, which makes an HTTP call to master, to fetch privileges. The Handler serving this HTTP Request uses a proxy cache - PrivilegeFetcherProxyService. This guy runs inside the master and keeps updating privileges, and serves the HTTP Requests to update privileges from program and system containers.

**Comment by anwar6953** [comment]
LGTM

**Comment by bdmogal** [comment]
Correction, CDAP-6943 will not be fixed by this PR. Will open an independent PR for that.

**Comment by bdmogal** [comment]
Will fix CDAP-6943 separately. Build is green for this one. Merging.

**Pull request merged**